### PR TITLE
refactor(Banner): Migrated from JS to TS

### PR DESCRIPTION
**Pull request by suvnshr**
+ Migrated `Banner` to Typescript
+ Migrated `BannerConstants` to Typescript
+ Migrated `banner.jest` to Typescript
+ Replaced type `string` with type `ReactNode` of `children` prop in `Button`
+ Added type `number` to `size` prop of `iconSubComponentProps`
+ Created a new type for `onClick` prop of `iconSubComponentProps`

Related issue: #894 

#### Basic
- [ ] Used plop (`npm run plop`) to create a new component.
- [x] PR has description.
- [ ] New component is functional and uses Hooks. 
- [ ] Component defines [`PropTypes`](https://reactjs.org/docs/typechecking-with-proptypes.html).
#### Style
- [ ] Styles are added to `NewComponent.modules.scss` file inside of the `NewComponent` folder.
- [ ] Component uses CSS Modules.
- [x] Design is compatible with [Monday Design System](https://design.monday.com/).
#### Storybook
- [ ] Stories were added to `/src/NewComponent/__stories__/NewComponent.stories.js` file.
- [ ] Stories include all flows of using the component.
#### Tests
- [x] Tests are compliant with [TESTING_README.md](TESTING_README.md) instructions.

### CODE REVIEW

**Comment by suvnshr** [comment]
@orrgottlieb Here you go :sparkle: 

Let me know if any changes are required.

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ const MIN_BUTTON_HEIGHT_PX = isIE11 ? 32 : 6;
 const UPDATE_CSS_VARIABLES_DEBOUNCE = 200;
 
 export interface ButtonProps {
-  children: string;
+  children?: React.ReactNode;
```
[the remark this answers is not in the document]
Comment (suvnshr) [reply src/components/Button/Button.tsx:23]: Thanks!

File: src/components/Icon/Icon.tsx [hunk 2]
```diff
@@ -8,15 +8,16 @@ import useIconProps from "./hooks/useIconProps";
...
 
 // eslint-disable-next-line no-unused-vars,@typescript-eslint/no-unused-vars,@typescript-eslint/no-empty-function
 const CLICK_NOOP = (event: React.MouseEvent) => {};
 
 interface iconSubComponentProps {
   ref?: Ref<any>;
   id?: string;
-  size?: string;
-  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
+  size?: string | number;
```
[the remark this answers is not in the document]
Comment (suvnshr) [reply src/components/Icon/Icon.tsx:19]: Thanks!

File: src/components/Banner/Banner.tsx [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+          {renderedTitle}
+          {renderedSubtitle}
+          {renderImage}
+        </div>
+      </aside>
+    );
+  }
+);
+
+// Banner.propTypes = {
```
[the remark this answers is not in the document]
Comment (suvnshr) [reply src/components/Banner/Banner.tsx]: Done.

File: src/components/Banner/Banner.tsx [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+  /**
+   * sub title value
+   */
+  subtitle?: string;
+
+  /**
+   * Add X button to the component when initialized and called when the button is clicked
+   */
+  onClose?: (event: React.MouseEvent<HTMLButtonElement>) => void;
```
[the remark this answers is not in the document]
Comment (suvnshr) [reply src/components/Banner/Banner.tsx]: Thanks, done.

**Comment by suvnshr** [comment]
@orrgottlieb I've made the required changes.

**Review by orrgottlieb — APPROVED** [review]
Thanks for an awesome job! and thank you  @suvnshr for the contribution, we will merge and add the appropriate label

P.S
If you have any feedback on the package.- i would love it if you can write it to me 
thank you

**Comment by suvnshr** [comment]
Thanks for the kind words @orrgottlieb 

overall I liked the design system.

Wanted to ask you whether the  `Banner` component is still in development?

coz I didn't see its doc in the storybook 

Thanks again!

**Pull request merged**